homeassistant/components/cover/reproduce_state.py

```diff
@@ -3,38 +3,164 @@
 from __future__ import annotations
 
 import asyncio
-from collections.abc import Iterable
+from collections.abc import Coroutine, Iterable
+from functools import partial
 import logging
-from typing import Any
+from typing import Any, Final
 
 from homeassistant.const import (
     ATTR_ENTITY_ID,
+    ATTR_SUPPORTED_FEATURES,
     SERVICE_CLOSE_COVER,
     SERVICE_CLOSE_COVER_TILT,
     SERVICE_OPEN_COVER,
     SERVICE_OPEN_COVER_TILT,
     SERVICE_SET_COVER_POSITION,
     SERVICE_SET_COVER_TILT_POSITION,
 )
-from homeassistant.core import Context, HomeAssistant, State
+from homeassistant.core import Context, HomeAssistant, ServiceResponse, State
+from homeassistant.util.enum import try_parse_enum
 
 from . import (
     ATTR_CURRENT_POSITION,
     ATTR_CURRENT_TILT_POSITION,
     ATTR_POSITION,
     ATTR_TILT_POSITION,
     DOMAIN,
+    CoverEntityFeature,
     CoverState,
 )
 
 _LOGGER = logging.getLogger(__name__)
 
-VALID_STATES = {
-    CoverState.CLOSED,
-    CoverState.CLOSING,
-    CoverState.OPEN,
-    CoverState.OPENING,
-}
+
+OPENING_STATES = {CoverState.OPENING, CoverState.OPEN}
+CLOSING_STATES = {CoverState.CLOSING, CoverState.CLOSED}
+VALID_STATES: set[CoverState] = OPENING_STATES | CLOSING_STATES
+
+FULL_OPEN: Final = 100
+FULL_CLOSE: Final = 0
+
+
+def _determine_features(current_attrs: dict[str, Any]) -> CoverEntityFeature:
```

Comment:
Note that this is required for the backwards compat.

To verify I checked out the test before any changes and ran it and verified the only change the was additional tilt close call that was what originally spawned looking into this.